types/node/index.d.ts

```diff
@@ -797,82 +809,93 @@ declare module "http" {
-        write(str: string, encoding?: string, fd?: string): boolean;
+        constructor();
 
-        addTrailers(headers: ServerResponseHeaders): void;
-        finished: boolean;
+        setTimeout(msecs: number, callback?: () => void): this;
+        destroy(error: Error): void;
+        setHeader(name: string, value: number | string | string[]): void;
         getHeader(name: string): number | string | string[] | undefined;
-        getHeaderNames(): string[]
-        getHeaders(): any
-        hasHeader(name: string): boolean
-        headersSent: boolean;
+        getHeaders(): HttpHeaders;
+        getHeaderNames(): string[];
+        hasHeader(name: string): boolean;
         removeHeader(name: string): void;
-        sendDate: boolean;
-        setHeader(name: string, value: string | string[]): void;
-        setTimeout(msecs: number, callback: Function): ServerResponse;
+        addTrailers(headers: HttpHeaders | [string, string][]): void;
+        flushHeaders(): void;
+
+        /**
+         * @deprecated OutgoingMessage.flush is deprecated. Use flushHeaders instead: DEP0001
+         */
+        flush(): void;
+    }
+
+    // https://github.com/nodejs/node/blob/master/lib/_http_server.js#L108-L256
+    export class ServerResponse extends OutgoingMessage {
         statusCode: number;
         statusMessage: string;
-        write(chunk: any, encoding?: string): any;
-        writeContinue(): void;
-        writeHead(statusCode: number, reasonPhrase?: string, headers?: ServerResponseHeaders): void;
-        writeHead(statusCode: number, headers?: ServerResponseHeaders): void;
 
-        // Extended base methods
-        end(): void;
-        end(buffer: Buffer, cb?: Function): void;
-        end(str: string, cb?: Function): void;
-        end(str: string, encoding?: string, cb?: Function): void;
-        end(data?: any, encoding?: string): void;
-    }
-    export interface ClientRequest extends stream.Writable {
-        // Extended base methods
-        write(buffer: Buffer): boolean;
-        write(buffer: Buffer, cb?: Function): boolean;
-        write(str: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, fd?: string): boolean;
+        constructor(req: IncomingMessage);
+
+        assignSocket(socket: net.Socket): void;
+        detachSocket(socket: net.Socket): void;
+        // https://github.com/nodejs/node/blob/master/test/parallel/test-http-write-callbacks.js#L53
+        // no args in writeContinue callback
+        writeContinue(callback?: Function): void;
+        writeHead(statusCode: number, reasonPhrase?: string, headers?: HttpHeaders): void;
+        writeHead(statusCode: number, headers?: HttpHeaders): void;
+
+        /**
+         * @deprecated Docs-only deprecated: DEP0063
+         */
+        writeHeader(statusCode: number, reasonPhrase?: string, headers?: HttpHeaders): void;
```

Comment:
I would not add deprecated APIs (in special as they are no longer in the docs) as this "invites" to us it. Same is valid for other internal/private properties.
But this is just my vote.